### DISCUSSION TITLE
How "weird" does my PHP installation need to be?

### DISCUSSION THREAD

**Discussion by fideloper** [php/frankenphp#54] [General]
The compiling from source section uses some flags for compiling PHP.

How far off from "standard php" are those? (e.g. installing PHP via `apt-get` in an Ubuntu container/server).

It looks like perhaps two most "non-standard" things are that is uses thread-safe PHP and disables Opcache JIT?

Are these specific to Worker mode?

Let me know if I can clarify anything in my question (and let me know if my ignorance here is causing me to ask a bad question :D )

**Comment by jaydrogers** [comment]
I am interested in hearing more about this too. I have Docker-PHP images that are optimized for WordPress and Laravel (which use PHP-FPM). I would love to explore offering FrankenPHP as an option too https://github.com/serversideup/docker-php/tree/dev

**Comment by fideloper** [comment]
One other way to ask this question is:

How do I make a container image using this that isn't 2.6G in size 😂

That must be related to all the dev dependencies used? (And perhaps `bullseye` is a large base image?)

```bash
docker image ls | grep franken

> dunglas/frankenphp   latest   a5e29aff03aa   9 days ago      2.67GB
```

**Reply by withinboredom** [reply]
<img width="1020" alt="image" src="https://user-images.githubusercontent.com/1883296/198393938-e2dd9939-9b5d-4cb8-a17e-1f7e56f88572.png">

They are much, much smaller now ;)

**Comment by dunglas** [comment]
The latest version is compatible with mainline PHP 8.2 (dev version, not the RCs yet) as long as it is compiled with the `--enable-zts` and `--disable-zend-signals` flags.

Docker images have been recently revamped (still building for `main`)  to be based on official images and reduce the final image size.

When https://github.com/docker-library/php/pull/1331 will be merged (if it is), we should be able to remove our custom stage and just build Caddy/FrankenPHP with the official images.

**Reply by withinboredom** [reply]
> When https://github.com/docker-library/php/pull/1331 will be merged (if it is), we should be able to remove our custom stage and just build Caddy/FrankenPHP with the official images.

Even if it isn't merged into the Docker library, we can still rebuild from the sources already in the official image. Most of the image bloat comes from that stage and moving libs between stages (so you have both versions of the files in the image instead of just one).

Either way, it should cut the build time tremendously since we won't have to compress the PHP sources... which currently is the most processor-intensive part of the entire build.

**Comment by back-2-95** [comment]
Images are getting smaller and eventually using FrankenPHP image will be like described here: https://github.com/dunglas/frankenphp/blob/main/docs/docker.md
